File: 0575-distribute-candies/0575-distribute-candies.js
/**
 * @param {number[]} candyType
 * @return {number}
 */
var distributeCandies = function(candyType) {
    let uniqueCandies = new Set(candyType).size;
    let upperLimit = candyType.length / 2;
    
    return Math.min(uniqueCandies, upperLimit);
};

File: 0575-distribute-candies/0575-distribute-candies.py
class Solution:
    def distributeCandies(self, candyType: List[int]) -> int:
        unique_candies = set(candyType)
        max_candies_to_eat = min(len(unique_candies), len(candyType) // 2)
        return max_candies_to_eat
                
            